text/0000-trinary-trytes.md

```diff
@@ -0,0 +1,180 @@
++ Feature Name: `trinary-trytes`
++ Start Date: 2019-09-09
++ RFC PR: [iotaledger/bee-rfcs#0000](https://github.com/iotaledger/bee-rfcs/pull/4)
++ Bee issue: [iotaledger/bee#0000](https://github.com/iotaledger/bee/issues/50)
+
+# Summary
+
+Define type `Trytes` for higher level usage. `Trytes` is going to be similar to
+`String` which is a vector of `u8` under the structure. It provides almost same
+methods but different signatures compares to `String`, because we are going to
+make sure every inputs to `Trytes` are valid **tryte alphabete**. **Tryte alphabete**
+is a table to make trytes easy to read. Since it serves this purpose and would
+interact with underlying ternary operations in most cases, `Trytes` will behave
+like string literal and each charaters are actually UTF-8 encoded character. So
+`9` has value of 57 and `A` to `Z` have values of `65` to `90` in its vector.
+
+# Motivation
+
+The reason to implement trytes only at the moment is not only to be loosly
+coupled from **trits**, but also layout what a ternary type could looks like. We
+will eventually have various encoding techniques. It may become a mess if we
+implement all in one structure, or seperate files but without a well-defined layout.
+For the reason to use vector of `u8` is because it's sufficent in most cases,
+implementations of `trit` encoding would need `bytes` from tokio or maybe
+`bitflags` instead. I would like to make it be able to represent as string
+literals and slice of values at the same time. As for choice of tryte alphabete
+being UTF-8 encoded is because we expect most users would like to conver strings
+and characters to `Trytes`. It makes more scene to stay same encoding as others
+in Rust. Maybe we will need to convert byte literals in the end, but still this
+can be done afterwards.
+
+# Detailed design
+
+## Trytes
+
+Type `Trytes` is a vector of valus `Vec<u8>` and moreover a smart pointer.
+The structure simply looks like this:
+
+```rust
+pub type Tryte = u8;
+
+pub struct Trytes {
+    vec: Vec<Tryte>,
```

Comment:
Do you plan to add more fields to this struct? Otherwise I would probably use the (`newtype`)[https://doc.rust-lang.org/1.0.0/style/features/types/newtype.html] pattern, i.e. a tuple struct:

```rust
struct Tytes(Vec<Trte>)
```

Reply:
Thanks, I will implement like this instead.